addons/addon-base-raas/packages/base-raas-post-deployment/lib/steps/create-service-catalog-portfolio.js

```diff
@@ -295,13 +300,55 @@ class CreateServiceCatalogPortfolio extends Service {
         DisableTemplateValidation: true,
         Type: 'CLOUD_FORMATION_TEMPLATE',
         Description: productToCreate.description || autoCreateDesc,
+        Name: nextProvArtifact,
       },
       ProductId: productId,
     };
     const data = await servicecatalog.createProvisioningArtifact(params).promise();
     return data.ProvisioningArtifactDetail.Id;
   }
 
+  async getNextArtifactVersion(latestVersionInSC) {
+    // For product artifacts in Service Catalog, versions and names are the same thing
+    // If no previous version found, assign 'v2' (this will help in next update cycle)
+    let returnVal = 'v2';
+    // Only finds version strings that match patterns:
+    // "V<n>", "V<n>.0", "V<n>.0.0", "v<n>", "v<n>.0", "v<n>.0.0"
+    const pattern = /^(v|V)(\d+\.)?(\d+\.)?(\*|\d+)$/;
+    let latestVersionNo;
+    if (latestVersionInSC) {
+      const parsedOutput = latestVersionInSC.match(pattern);
+      this.log.info(`${parsedOutput}`);
+      if (parsedOutput && parsedOutput.length > 1) {
+        const longestMatch = parsedOutput.sort(function(a, b) {
+          return b.length - a.length;
```

Comment:
A lot of efforts are made to accommodate different type of version strings "V<n>", "V<n>.0", "V<n>.0.0", "v<n>", "v<n>.0", "v<n>.0.0". Since these products are created and updated by the system, can we define the version number to only follow one convention? Maybe v1, v2, v3? This way we can avoid the complication accommodating different dots and digits.

Reply:
We initially decided to have the initial product version to be "V1.0.0", but realized the minor and patch versions don't really make sense in template versions and furthermore difficult to determine if upcoming changes are minor or patches (thereby rendering them useless).

Now we have changed `autoCreateVersion` (assigned to the first version of the product) to "v1". The regex pattern is intentionally kept to match both the previous version string format (eg. V1.0.0) as well as the new ones (eg. v1). In the long run, new product versions would stick to the "v<n>" format.

Reply:
You may want to add a summary of this as a comment in code. Just explaining that it supports "legacy" versions  like v.1.2.3 but that new versions are simplified to look like v2